### PR TITLE
Small percentage of roundstart cameras deactivated

### PR DESCRIPTION
**Pull request by beesil**
[tweak][balance]
Makes a random ~~fifteen~~ ten percent of roundstart cameras deactivated. I'm not dead set on ~~fifteen~~ ten percent so feel free to suggest higher/lower percentages as you see fit. 
Additionally adds a flawless camera subtype with a zero percent deactivation chance.

The intent of this is to discourage meta knowledge that cut cams = antag activity, in the same vein as randomized suit sensors. 

:cl:
 * tweak: Ten percent of roundstart cameras are now randomly deactivated.

### CODE REVIEW

**Comment by Lokiamis** [comment]
That's like 30 cameras on Boxstation man

**Comment by bathosbathtime** [comment]
The AI nerf we all need

**Comment by aceedex** [comment]
Im no byond genius, but doesnt that simply give  a camera a 15% chance to deactivate on startup meaning its stupidly susceptible to RNG fuckery.

Aside from that, I dont think that's needed. Plenty of places already start without cameras and having a camera cut ISNT a tell tale sign of antag fuckery. I do it all the time as non antag mechanic. Plus, if you deactivate a camera hiding what actual type of activity goes on in there is enough.

**Comment by ghost** [comment]
I like this a lot.

**Comment by ShiftyRail** [comment]
Okay but not all cameras should be equal
Armoury cameras are more important than a random one in bumfuck nowhere

**Comment by Lokiamis** [comment]
This would disable 41 cameras on Boxstation. This is retarded.

**Comment by ShiftyRail** [comment]
@Lokiamis How many cameras are there on Box ? I can actually do the maths for it with stats and shit.

**Comment by agentofzog2** [comment]
I feel like this is too box centric (and meta centric I guess but fuck if I know when that map will be back) since deff and roid actually have decent places to hide out in that won't be discovered by a random assistant.

Also will this affect outpost cams and those random cams in certain vaults?

**Comment by Lokiamis** [comment]
@ShiftyRail There are 270 listed by the camera console. I am aware there are some tiles with 2 cameras for whatever reason so it's slightly higher.

**Comment by beesil** [comment]
@Lokiamis 
>That's like 30 cameras on Boxstation man

Testing once, there were 34 deactivated cameras. Like I said though I'm not opposed to making the percentage lower.

@ShiftyRail 
>Okay but not all cameras should be equal
> Armoury cameras are more important than a random one in bumfuck nowhere

When it's not waaay past my bedtime I can look into making AI core/armory/etc cameras whitelisted.

**Comment by Lunstra** [comment]
The number is a touch too high, maybe 10% would be better, and certain high risk cameras should be excluded like the bridge, AI core, armoury and such, but the concept is sound because fuck how powerful the omniscient eye in the sky valid alarm is

**Comment by LordMyroc** [comment]
15% is way, way too high a number, but I like the idea itself. I think even 5% is sufficient. There are a LOT of cameras on the station.

**Comment by DrSnips** [comment]
My big issue is that it really stunts the usefulness of an AI especially if you're a part of the grey tide who uses LAW 2 OPEN THE DOOR GET ON THE FLOOR EVERYONE DO THE DINOSAUR. On top of that this leads to a ton of dead zones letting AI's have a lot of plausible deniability for doing their job or obeying laws. "Sorry admins I thought my ones human was off station when I decided to make it bacon since I couldn't see them on the station anywhere for a long time now:^)"

**Comment by iFlashYou** [comment]
based

**Comment by soyconsumer** [comment]
blessed PR

**Comment by Exxion** [comment]
15% does seem too high, but honestly I doubt there would be that many dead zones past what are already there
Camera areas overlap, so the percentage of the area that would lose coverage would be less than the percentage of cameras disabled

**Comment by jknpj** [comment]
make it a proc so we can override it later for flawless cameras

**Comment by boy2mantwicethefam** [comment]
Leave it as it is, it gives engies something to do, namely actually repairing stuff

**Comment by ShiftyRail** [comment]
Mathematically the situation is about `n` independent trials with a `p` chance of succeeding every time.
It's called a binomial distribution : https://en.wikipedia.org/wiki/Binomial_distribution

The moments of which are well-known : 
```
Mean: np = 0.15*270 = 40.5
Mode (the number that'll appear the most when doing simulations) : (n+1)*p rounded down = 40
Median (exactly half of rounds will have less cameras disabled, exactly half of rounds will have more cameras disabled) = 41

Variance : n*p*(1-p) = 270*0.15*0.85 = 34

Which gives a "standard variation" of 6.
```

95% of rounds will have a number of cameras disabled between 29 and 52.
66% of rounds will have a number of cameras disabled between 35 and 47.

That's just the numbers

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -41,6 +41,10 @@ var/list/camera_names=list()
 	var/vision_flags = SEE_SELF //Only applies when viewing the camera through a console.
 	var/health = CAMERA_MAX_HEALTH
 
+/obj/machinery/camera/initialize()
```
Comment (DDihanov) [inline code/game/machinery/camera/camera.dm:48]: Where does this method get called again?

Comment (ShiftyRail) [reply code/game/machinery/camera/camera.dm:48]: When the game starts, it loops through all atoms in the world and calls their `initialize()`

Comment (DDihanov) [reply code/game/machinery/camera/camera.dm:48]: Never mind I found it: https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/controllers/subsystem/objects.dm

Comment (jknpj) [reply code/game/machinery/camera/camera.dm:48]: roundstart

**Comment by brndd** [comment]
I agree 15% is too high, but this is a good change because it gives crew members plausible deniability if they intentionally disable the camera in some room for eg. cult activities. The AI won't immediately know that it was broken on purpose.

There's however the problem that humans will likely be reluctant to fix these, because they only concern the AI. AIs will probably have to get one of their borgs (assuming they *have* borgs) to go engieborg to fix all the cameras or just deal with it.

**Comment by agentofzog2** [comment]
> 95% of rounds will have a number of cameras disabled between 29 and 52.
> 66% of rounds will have a number of cameras disabled between 35 and 47.
> 


FUUUUUUUUUUUUCK THAAAAAAAAAAAAAAAT I'm not fixing all that shit
29 should be the maximum not the minimum.

Also think of the poor malfs

**Comment by Caboozles** [comment]
10% with some high priority places?
IE, Armor, Captain’s Quarters, AI core.

**Comment by Allspice** [comment]
Box armoury has 3 cameras with high redundancy in coverage. Cap office and upload only have one, though.

**Comment by brndd** [comment]
Yeah it should probably be like 10-15, 20 tops, cameras broken in the average round rather than fucking 40.

**Comment by Allspice** [comment]
not sure that 40 is the scary number that it's being made out to be due to the redundancy in other areas as well such as the halls, and fixing 40 frinking cameras would happen less than an engi being maybe asked to fix a couple cameras in high-value areas.

would be nice to get three or so example images for how it looks on box from an AI perspective.

**Comment by Kitfox88** [comment]
this is fuckin' retarded, have it hard do 1-5 if you're deadset on it, not fucking 15% when every station has dozens if not hundreds

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Remember that we already have randomly broken lightbulbs and I can't remember a single time where a janitor was successfully made to change a lightbulb

And lights affect everyone, not just 1 role and maybe ostensibly the warden

What I'm saying is that no, engineers won't realistically be asked to fix cameras and if they are realistically they won't do it anyways

**Comment by GoodG1448** [comment]
I have mixed feeling about this. I see people talking about box but what about packet/roid/deff total camera count?. What if every department had flawless cameras in key areas but had other non-flawless ones off at roundstart. Like 1-2 deactivated cameras per department at roundstart.

**Comment by beesil** [comment]
Changed normal cameras to have a 10% chance to deactivate, and created a subtype with a 0% chance. 

Testing three times with a 10% chance, there were 24, 28, and 27 deactivated cameras.

**Comment by SonixApache** [comment]
how many broken lights are there at roundstart again?

**Comment by DeityLink** [comment]
That's the cult buff we needed all along. 10% is way too high however. I think we should start with 4%.

**Comment by Killette2** [comment]
5%

**Comment by Durgen** [comment]
@otatoh make cams in the core and upload immune to the RNG

**Comment by GoodG1448** [comment]
A range between 4% and 6% seems optimal.

**Comment by SGuivenne** [comment]
If you're going this route you should have some sort of time delay "faulty' cameras.  Ones that will randomly cut out or go out of focus at some point in the round.

**Comment by Durgen** [comment]
Can we make the disable % a rollable number between 0 and 15

**Comment by brndd** [comment]
> Can we make the disable % a rollable number between 0 and 15

It kind of already is rollable because every camera individually has a 10% chance to be disabled, rather than 10% of cameras on station being disabled.

**Comment by beesil** [comment]
Regarding percentage, 5% is too low to make a difference. In a test, 5% failure chance deactivated 12 cameras, most of which's area covered overlapped with other cameras. This isn't counting the 13 proposed flawless cameras either, which will make the number even lower.

@Durgen 
Planning to make the AI core, armory, bridge, and captain's office cameras flawless in another PR to avoid map conflict gas.

**Comment by brndd** [comment]
Another idea would be to hardcode an area whitelist where cameras can never be randomly disabled. This would let you avoid the mapping gas, and would make sure future mappers don't fuck things up by forgetting to use the unfailing camera subtype.

**Comment by TurboGay** [comment]
This is a fairly significant nerf to malfunctioning AI, given that you usually rely on hacking out of the way APC's in rooms that are seldom traversed. Not sure how hard it would be to code but either an exception for malf or a module that they can purchase to re-enable all faulty cameras would be nice.

**Comment by brndd** [comment]
Just watch out it doesn't turn into ";THE CAMERAS ARE ALL FIXED GUYS IT'S MALF"

**Comment by TurboGay** [comment]
> 
> 
> Just watch out it doesn't turn into ";THE CAMERAS ARE ALL FIXED GUYS IT'S MALF"

Yeah, that would be pretty bad. Probably even harder to code but if it worked like the overload module in that you get charges and can pick and choose the cameras you want to re-enable or from a list that would be a workaround to it becoming a malf meta check.

**Comment by Hoshi-chan** [comment]
>OPEN THE FUCKING DOOR AI

>Sorry, the camera's off. I can't see you, human.

**Comment by Kurfursten** [comment]
This should be carefully monitored. In seven days, on the 20th, ping me directly with findings related to how this has affected rounds. At that time if adjustments are needed, we can tweak up or down. I think this is really one of those PRs people are going to have to experience to make a call on where the % chance falls. I wouldn't mind a failsafe, too, like a once over check at the end of initialization to prevent the one in a million round where dice gods give us 180 cameras disabled.

But the votes are pretty clear on this point. We'll monitor and discuss this thoroughly and it will be a community effort.

**Pull request merged**